### ISSUE TITLE
Color-contrast inconsistency of bgOverlap message

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3309]
The code to detect overlap only works when an element is scrolled into view. This isn't at all necessary, and is a hold-over from before axe-core was using a grid to test color-contrast. You can tell by looking at something like the following, which reports `#target` as overlapped when scrolled into view, but ignores the overlapping element when it's not in view.

```html
<div style="height: 1em; overflow: auto; background: cyan;">
  <div style="background: rgba(0,255,255,0.6); margin-bottom: -1em; height: 1em; position:relative;"></div>
  <div id="target">foo</div>
</div>
```

I haven't seen this produce false positives, but in theory it can. The more problematic part is that it changes the result, based on where in the page the scroll happens.

**Comment by padmavemulapati** [comment]
Verified with the latest dev branch code,
Seeing no failures on `color-contrast`
when `target` as id and overlapped it ignores the overlapping element and also for inline elements with position absolute, these two cases show it as `incomplete`
otherwise:
non-opaque is obscuring it, scrolled out of view, and nested inline elements in the middle of a text show in `scrollable-region-focusable` error not `color-contrast`
![image](https://user-images.githubusercontent.com/47409579/147907560-902ff468-28da-4620-914e-2d2dd959f20a.png)
